source/include/DebuggingContext.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 National Instruments
+// SPDX-License-Identifier: MIT
+
+/*! \file
+	\brief Tools to implement a debugging context
+ */
+
+#include "TypeAndDataManager.h"
+#include "map"
+namespace Vireo
+{
+class DebuggingContext
+{
+ private:
+    std::map<SubString, bool, CompareSubString> _debugPointState;
+ public:
+    bool GetDebugPointState(SubString objectID)
+    {
+         typedef std::map<SubString, bool>::iterator iterator;
+         iterator it = _debugPointState.find(objectID);
+
+         if (it == _debugPointState.end()) {
+         // error out:  std::cout << "Key-value pair not present in map";
+              return false;
+         }
+        return it->second;
+    }
+    void SetDebugPointState(SubString objectID, bool state)
+    {
+         _debugPointState[objectID] = state;
+    }
```

Comment:
I would prefer to move these to the cpp file. Header file should ideally contain only single line function definitions.

Reply:
Currently, it was showing some compile-time error while moving to cpp file. I will fix it later.